tools/install/scripts/writePackageNames.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-console */
+import { formatLG } from '@lg-tools/lint';
+import { getAllPackageNames } from '@lg-tools/meta';
+import fse from 'fs-extra';
+import path from 'path';
+
+const PACKAGE_NAMES_FILE_PATH = '../src/ALL_PACKAGES.ts';
+
+writeAllPackageNames();
+
+/**
+ * Scan the repository for all leafygreen packages and write their names to a static file
+ * This file will be used by the install script instead of fetching from NPM
+ * @returns The array of package names that were written to the file
+ */
+async function writeAllPackageNames() {
+  try {
+    // Get all package names from the repository
+    const packageNames = getAllPackageNames();
+
+    // Make sure we have at least some packages
+    if (!packageNames.length) {
+      console.warn('No packages found in the repository');
+      return [];
+    }
+
+    console.log(`Found ${packageNames.length} packages in the repository`);
+
+    // Write the package names to a static file as JSON
+    const filePath = path.resolve(__dirname, PACKAGE_NAMES_FILE_PATH);
+
+    // Make sure the directory exists
+    const dirPath = path.dirname(filePath);
+
+    if (!fse.existsSync(dirPath)) {
+      fse.mkdirSync(dirPath, { recursive: true });
+    }
+
+    const packageNamesFileContents = `// This file is auto-generated by the writeAllPackageNames script
+// Do not edit this file directly
+
+export const ALL_PACKAGES = ${JSON.stringify(packageNames, null, 2)};
+`;
+
+    const formattedContents = await formatLG(
+      packageNamesFileContents,
+      filePath,
+    );
+    // Write the file
+    fse.writeFileSync(filePath, formattedContents);
+
+    console.log(
+      `${packageNames.length} package names written to ${PACKAGE_NAMES_FILE_PATH}`,
+    );
+  } catch (error) {
+    console.log('Error writing package names to static file:', error);
```

Comment:
nit: Would it be beneficial to add the name of the file path?

Reply:
I don't think so? It's a hard coded file name, so it's not that useful imo